### PR TITLE
Wk/taco 363 support http non-tls proxied backend(s)

### PR DESCRIPTION
**Pull request by manimaul**

### CODE REVIEW

File: int-tests/server_controller.py [hunk 1]
```diff
@@ -69,7 +69,9 @@ def __init__(self, script, ready_str, *args, **kwargs):
     self._name = kwargs.get('name', 'unnamed')
     self._verbose = kwargs.get('verbose', False)
     if len(args) > 1:
-      argv = functools.reduce(lambda a, b: ("", str(a).strip() + " " + str(b).strip()), args)[1]
+      argv = ' '
+      for each in [str(i).strip() for i in args if len(str(i)) > 0]:
+        argv += ' ' + each
```
Comment (pdex) [inline int-tests/server_controller.py]: I always cry a little bit when a lambda dies, but this list comprehension is more readable.

File: xio-core/src/main/java/com/xjeffrose/xio/http/ClientChannelInitializer.java [hunk 2]
```diff
@@ -55,8 +55,8 @@ protected void initChannel(Channel channel) throws Exception {
       // No need for an http2 handler as we don't allow that over cleartext
       channel
           .pipeline()
-          .addLast("codec", new HttpServerCodec())
-          .addLast("application codec", new Http1ServerCodec());
```
Comment (manimaul) [inline xio-core/src/main/java/com/xjeffrose/xio/http/ClientChannelInitializer.java]: whoops :)

File: xio-core/src/main/java/com/xjeffrose/xio/http/ClientChannelInitializer.java [hunk 3]
```diff
@@ -55,8 +55,8 @@ protected void initChannel(Channel channel) throws Exception {
...
+          .addLast("codec", new HttpClientCodec())
+          .addLast("application codec", new Http1ClientCodec());
```
Comment (pdex) [inline xio-core/src/main/java/com/xjeffrose/xio/http/ClientChannelInitializer.java]: 🤦‍♂️ 🤦‍♀️

Comment (pdex) [reply xio-core/src/main/java/com/xjeffrose/xio/http/ClientChannelInitializer.java]: I'm guessing I did that.

File: .circleci/config.yml [hunk 4]
```diff
@@ -13,10 +13,10 @@
         - v1-dependencies-
     - run:
         name: install dependencies
-        command: ./gradlew :xio:dependencies
+        command: ./gradlew dependencies
     - run:
         name: run build test verify
-        command: ./gradlew :xio:build
```
Comment (manimaul) [inline .circleci/config.yml]: we want to run the `ReverseProxyFunctionalTest` and others (?) which is/are not in the xio gradle module

**Pull request merged**